lib/srv/heartbeatv2.go

```diff
@@ -76,29 +75,29 @@ func NewSSHServerHeartbeat(cfg SSHServerHeartbeatConfig) (*HeartbeatV2, error) {
 	}
 
 	var metadataPtr atomic.Pointer[metadata.Metadata]
-	var once sync.Once
+	var metadataFetched atomic.Bool
```

Comment:
Can you do this without two separate atomics? When `metadataPtr` holds nil, you know you need to retry. If it holds a non-nil pointer, then you're done.